src/lib/common/sol-platform-linux-common.c

```diff
@@ -263,3 +307,581 @@ sol_platform_impl_get_os_version(char **version)
+            goto err;
+        }
+    }
+    endmntent(tab);
+    return 0;
+
+err:
+    SOL_PTR_VECTOR_FOREACH_IDX (vector, mp, i) {
+        sol_mount_point_free(mp);
+    }
+    sol_ptr_vector_clear(vector);
+    return -ENOMEM;
+}
+
+int
+sol_platform_impl_get_mount_points(bool hplug_only, struct sol_ptr_vector *vector)
+{
+    struct sol_ptr_vector mtab, sol_mtab, delete;
+    struct sol_mount_point *mp, *mp2;
+    uint16_t i, j;
+    int err;
+
+    if (!hplug_only) {
+        return parse_mount_point_file(SYS_MTAB_FILE, vector);
+    }
+
+    sol_ptr_vector_init(&mtab);
+    err = parse_mount_point_file(SYS_MTAB_FILE, &mtab);
+    SOL_INT_CHECK_GOTO(err, < 0, err_mtab);
+
+    sol_ptr_vector_init(&sol_mtab);
+    err = parse_mount_point_file(SOL_MTAB_FILE, &sol_mtab);
+    SOL_INT_CHECK_GOTO(err, < 0, err_sol_mtab);
+
+    sol_ptr_vector_init(&delete);
+    SOL_PTR_VECTOR_FOREACH_IDX (&sol_mtab, mp, i) {
+        SOL_PTR_VECTOR_FOREACH_IDX (&mtab, mp2, j) {
+            if (streq(mp2->dev, mp->dev) && streq(mp2->mpoint, mp->mpoint)) {
+                sol_ptr_vector_append(vector, mp2);
+                sol_ptr_vector_remove(&mtab, mp2);
+            } else {
+                sol_ptr_vector_append(&delete, mp2);
+            }
+        }
+        sol_mount_point_free(mp);
+    }
+
+    SOL_PTR_VECTOR_FOREACH_IDX (&delete, mp, i) {
+        sol_mount_point_free(mp);
+    }
+
+    sol_ptr_vector_clear(&delete);
+    sol_ptr_vector_clear(&sol_mtab);
+    sol_ptr_vector_clear(&mtab);
+
+    return 0;
+
+err_sol_mtab:
+    sol_ptr_vector_clear(&sol_mtab);
+err_mtab:
+    sol_ptr_vector_clear(&mtab);
+    return err;
+}
+
+static int
+sol_mtab_remove_entry(const char *mpoint, struct sol_buffer *output)
+{
+    int fd = 0;
+    struct sol_vector lines;
+    struct sol_str_slice content = SOL_STR_SLICE_EMPTY;
+    struct sol_str_slice *itr;
+    struct sol_file_reader *reader;
+    uint16_t idx;
+
+    fd = open(SOL_MTAB_FILE, O_RDONLY | O_CLOEXEC);
+    if (!fd) {
+        SOL_ERR("Could not open "SOL_MTAB_FILE " - %s",
+            sol_util_strerrora(errno));
+        return -errno;
+    }
+
+    reader = sol_file_reader_fd(fd);
+    if (!reader) {
+        SOL_ERR("Could not read "SOL_MTAB_FILE " file - %s",
+            sol_util_strerrora(errno));
+        return -errno;
+    }
+
+    content = sol_file_reader_get_all(reader);
+    lines = sol_util_str_split(content, "\n", 0);
+    SOL_VECTOR_FOREACH_IDX (&lines, itr, idx) {
+        char buff[512];
+        snprintf(buff, itr->len, "%s", itr->data);
+        if (!strstr(buff, mpoint))
+            sol_buffer_append_printf(output, "%s", buff);
+    }
+
+    sol_vector_clear(&lines);
+    sol_file_reader_close(reader);
+    close(fd);
+    return 0;
+}
+
+// since we don't have a mtab entry remove function we must implement it ourselves
+static int
+sol_mtab_cleanup(const char *mpoint)
+{
+    int res, err, fd;
+    struct sol_buffer output;
+
+    res = fd = 0;
```

Comment:
If you must initialize `fd` for some reason, initialize it with an invalid value, such as `-1`. `0` is `stdin` usually.